apps/frontend/src/pages/CreateCharity.tsx

```diff
@@ -0,0 +1,651 @@
+      setSent(false);
+    }
+  };
+
+  // Uploads an array of files to the blobstore
+  const asyncUploadImages = async (
+    Images: ImageUploadObject[],
+  ): Promise<ImageUploadedObject[] | false> => {
+    const retrievedImages: ImageUploadedObject[] = [];
+    const uploadPromises = Images.map(async (image) => {
+      try {
+        const imageFile = image.image;
+        // Upload the image
+        const response = await _axios_instance.post("/images", imageFile, {
+          headers: {
+            "Content-Type": "multipart/form-data",
+          },
+        });
+
+        return { url: response.data.url, index: image.index };
+      } catch (error) {
+        console.error("Error uploading image:", error);
+        return null;
+      }
+    });
+
+    try {
+      const results = await Promise.all(uploadPromises);
+
+      results.forEach((result) => {
+        if (result) {
+          retrievedImages.push(result);
+        }
+      });
+
+      // Return the list of successfully uploaded images
+      return retrievedImages;
+    } catch (error) {
+      console.error("Error in uploading image process:", error);
+      return false;
+    }
+  };
+
+  // Uploads the main charity image to the s3 blob store
+  const asyncListingImageWrapperLogo = async (): Promise<boolean | string> => {
+    try {
+      if (!logoImage) return true;
+      const logoImagesObjectArray = await asyncUploadImages([
+        { image: logoImage, index: 0 },
+      ]);
+
+      if (logoImagesObjectArray) {
+        const newURL = logoImagesObjectArray[0].url;
+        setNewCharityObject((prevState) => ({
+          ...prevState,
+          imageUrl: newURL,
+        }));
+
+        return newURL;
+      } else {
+        console.error("No images uploaded for logoImage.");
+        return false;
+      }
+    } catch (error) {
+      console.error("Error uploading logo image:", error);
+      return false;
+    }
+  };
+
+  // Uploads the partner images to the s3 blobstore and packages the partner logos
+  const asyncListingImageWrapperPartners = async (
+    logoUrlResult: string | boolean,
+  ): Promise<boolean | CharityObject> => {
+    try {
+      if (partnerImages.length === 0) {
+        return { ...newCharityObject, imageUrl: logoUrlResult as string };
+      }
+
+      // Check if newCharityObject.organizations is an array
+      if (!Array.isArray(newCharityObject.organizations)) {
+        console.error(
+          "newCharityObject.organizations is not an array:",
+          newCharityObject.organizations,
+        );
+        return false;
+      }
+
+      // Upload partnerImages
+      const partnerImagesArray = partnerImages
+        .filter((obj) => obj.image !== undefined)
+        .map((obj, index) => ({ image: obj.image as File, index }));
+
+      const partnerImagesObjectArray =
+        await asyncUploadImages(partnerImagesArray);
+      if (partnerImagesObjectArray && partnerImagesObjectArray.length > 0) {
```

Comment:
```suggestion
      if (partnerImagesObjectArray?.length > 0) {
```

Reply:
Linter doesn't like since I have that object as also possibly boolean